### PR TITLE
Upgrade depencency to ramsey/uuid 3.x

### PR DESCRIPTION
**Pull request by robertlemke**
This change bases the existing code on the 3.x version of ramsey/uuid (previously called rhumsaa/uuid) in order to avoid conflicts with third-party libraries which require version 3.0 or higher.

### CODE REVIEW

**Comment by codeliner** [comment]
Same like https://github.com/prooph/common/pull/52

File: tests/AggregateChangedTest.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ public function it_has_a_new_uuid_after_construct()
     {
         $event = AggregateChanged::occur('1', []);
 
-        $this->assertInstanceOf('Rhumsaa\Uuid\Uuid', $event->uuid());
+        $this->assertInstanceOf('Ramsey\Uuid\Uuid', $event->uuid());
```
Comment (malukenho) [inline tests/AggregateChangedTest.php]: `\Ramsey\Uuid\Uuid::class`

**Comment by oqq** [comment]
Maybe it should be better to move ramsey/uuid to require-dev part in composer.json, since it would´t be used in application source at all.

**Comment by sandrokeil** [comment]
@oqq Good catch! :+1: We've done it already in [prooph event-store](https://github.com/prooph/event-store/pull/166), but missed it here. Do you want to bring a PR please?

**Comment by oqq** [comment]
@sandrokeil 
sure https://github.com/prooph/event-sourcing/pull/30

**Comment by trq** [comment]
Just moving this into require-dev doesn't fix collisions with other packages. https://github.com/prooph/laravel-package for example is impossible to installcurrently.

```
  Problem 1
    - prooph/laravel-package v0.2 requires prooph/event-sourcing ^4.0 -> satisfiable by prooph/event-sourcing[v4.0].
    - Installation request for prooph/laravel-package ^0.2.0 -> satisfiable by prooph/laravel-package[v0.2].
    - Conclusion: remove ramsey/uuid 3.5.0
    - Conclusion: don't install ramsey/uuid 3.5.0
    - prooph/event-sourcing v4.0 requires ramsey/uuid ~2.8 -> satisfiable by ramsey/uuid[2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.9.0].
    - Can only install one of: ramsey/uuid[2.8.0, 3.5.0].
    - Can only install one of: ramsey/uuid[2.8.1, 3.5.0].
    - Can only install one of: ramsey/uuid[2.8.2, 3.5.0].
    - Can only install one of: ramsey/uuid[2.8.3, 3.5.0].
    - Can only install one of: ramsey/uuid[2.8.4, 3.5.0].
    - Can only install one of: ramsey/uuid[2.9.0, 3.5.0].
    - Installation request for ramsey/uuid (locked at 3.5.0) -> satisfiable by ramsey/uuid[3.5.0].
```

File: tests/AggregateChangedTest.php (hunk 1 above)
Comment (trq) [inline tests/AggregateChangedTest.php]: This namespace changes between 2.* and 3.* so you can't simply support "~2.8|~3.0". Can we just get it upgraded?

**Comment by oqq** [comment]
@trq please use https://github.com/MattKetmo/uuid-2x-bridge for this kind of collision.

**Comment by prolic** [comment]
@oqq is right! Use the uuid-bridge for now if you need to run prooph with ramsey/uuid v3.

@trq an upgrade to ramsey/uuid 3.x is only possible with next major release (which hopefully comes by the end of this year).

**Comment by trq** [comment]
Oh wow, thanks.

**Pull request closed without merging**